argo-data-submission-wf/main.nf

```diff
@@ -158,36 +181,91 @@ workflow ArgoDataSubmissionWf {
       )
       sequence_files=egaWf.out.sequence_files
     } else {
-      sequence_files=Channel.fromPath(file_info_tsv) | splitCsv( header : true , sep:'\t') | map( row -> file("${row.path}",checkIfExists : true))
+      sequence_files=Channel.fromPath(file_info_tsv) | splitCsv( header : true , sep:'\t') | map( row -> file("${row.path}",checkIfExists : true)) 
     }
     
-    // use seq-tools to validate payload and sequence data
-    valSeq(
+    // Split files into CRAM and nonCRAM files accordingly
+    cram_sequence_files=sequence_files.filter(row -> row =~ /cram$/)
+    not_cram_sequence_files=sequence_files.filter(row -> row =~ /bam$|gz$|bz2/)
+
+    // If reference genome is not provided...
+    if (ref_genome_fa.startsWith("NO_FILE")){
+      // Generate metadata payload per normal
```

Comment:
The workflow depends on the value of `ref_genome_fa` to decide whether do cram2bam or not.
Do you think it is better to have a dedicated param flag like `cram_support` to indicate that the input files contain CRAMs. So that we can check at the beginning:
- if `cram_support=true`, `cram_sequence_files` is not empty && `ref_genome_fa` is not `NO_FILE`
- if `cram_support=false`, `cram_sequence_files` is empty

Reply:
Might make the command unnecessarily bulky with the addition of both `ref_genome_fa` and a `cram_support` flag. 

Instead I can add the `ref_genome_fa` check earlier so that the pipeline will exit if `ref_genome_fa` does not exist and `CRAM` files are detected?

Reply:
Yeah, that's another solution. It's better to add some checks to avoid the scenario that the user forget to add `ref_genome_fa` or provide it without purpose, however this will mislead the running of the workflow, and failed at later steps.